sig-scalability/blogs/k8s-services-scalability-issues.md

```diff
@@ -0,0 +1,117 @@
+# Known Scalability Issues with Kubernetes Services
+
+_by Shyam JVS, Google Inc (with inputs from Brian Grant & Wojciech Tyczynski)_
+
+**June 2018**
+
+This document serves as a catalog of issues we've known/discovered with kubernetes services as of June 2018, focusing on their scalability/performance. The purpose of the document is to make the information common knowledge for the community, so we can work together towards improving it. Listing them below in no particular order.
+
+
+## Performance of iptables when large number of service ports exist
+
+#### Issue
+
+Iptables can be slow in the presence of a large number of services. As the number of service ports increases, the KUBE-SERVICES chain gets longer and since it’s also evaluated very frequently that can bring down the performance. Also, it seems that manipulating a chain with lots of rules (e.g. KUBE-SERVICES) takes noticeably longer than others.
```

Comment:
The second sentence seems to be talking about packets moving through the kernel being slowed down by the presence of too many iptables rules, but the third then switches to talking about the speed of making changes to the iptables rules, which is a different problem and probably belongs in the next section?

If this is supposed to be about the speed of packet processing, rather than the speed of "iptables-restore", then it might be good to clarify that in the section heading, since just "Performance of iptables" is ambiguous as to which part of the system it's referring to.

(FWIW https://github.com/kubernetes/kubernetes/pull/56164 / https://github.com/kubernetes/kubernetes/pull/57461 / https://github.com/kubernetes/kubernetes/pull/60306 are some examples of fixes specifically for network performance as opposed to reload speed. I'm not sure if there are any currently-open issues about iptables-related network performance problems.)

Reply:
You're right. Let me make the distinction clear.

Reply:
I've incorporated the above - PTAL. Also thanks for those really useful leads!